### PR TITLE
Replace ENDIAN macros

### PR DESCRIPTION
**Pull request by bernardnormier**
This PR replaces the ENDIAN macros in IceUtil/Config.h by the standard (C++20) std::endian enum.

### CODE REVIEW

File: cpp/include/Ice/OutputStream.h [hunk 1]
```diff
@@ -615,22 +615,7 @@ class ICE_API OutputStream : public IceInternal::Buffer
...
-        *dest = *src;
-#else
-        const std::uint8_t* src = reinterpret_cast<const std::uint8_t*>(&v);
-        *dest++ = *src++;
-        *dest++ = *src++;
-        *dest++ = *src++;
-        *dest = *src;
-#endif
-    }
+    void write(std::int32_t v, Container::iterator dest);
```
Comment (bernardnormier) [inline cpp/include/Ice/OutputStream.h:618]: Moved out of line

**Review by externl — APPROVED** [review]
Looks good!

File: cpp/src/Ice/CollocatedRequestHandler.cpp [hunk 2]
```diff
@@ -53,13 +54,15 @@ void
 fillInValue(OutputStream* os, int pos, int32_t value)
 {
     const uint8_t* p = reinterpret_cast<const uint8_t*>(&value);
-#ifdef ICE_BIG_ENDIAN
-    reverse_copy(p, p + sizeof(std::int32_t), os->b.begin() + pos);
-#else
-    copy(p, p + sizeof(std::int32_t), os->b.begin() + pos);
-#endif
+    if (endian::native == endian::big)
```
Comment (pepone) [inline cpp/src/Ice/CollocatedRequestHandler.cpp:57]: Can we use `constexpr` like in the example https://en.cppreference.com/w/cpp/types/endian?

File: python/modules/IcePy/Types.cpp [hunk 3]
```diff
@@ -1772,7 +1772,7 @@ IcePy::SequenceInfo::marshalPrimitiveSequence(const PrimitiveInfoPtr& pi, PyObje
 
             if(pi->kind != PrimitiveInfo::KindByte)
             {
-#   ifdef ICE_BIG_ENDIAN
+#   ifdef WORDS_BIGENDIAN
```
Comment (pepone) [inline python/modules/IcePy/Types.cpp:1775]: What is this macro?

**Review by pepone — APPROVED** [review]

File: python/modules/IcePy/Types.cpp (hunk 3 above)
Comment (bernardnormier) [reply python/modules/IcePy/Types.cpp:1775]: It's a C-Python macro, set to 1 on big endian systems.

**Pull request merged**